register.lic

```diff
@@ -0,0 +1,73 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#register?
+
+  Search a deed register for the specified crafting material.
+  Output all lines that match.
+=end
+
+custom_require.call(%w[common common-items])
+
+class Register
+  include DRC
+  include DRCI
+
+  def initialize
+    arg_definitions =
+      [
+        [
+          { name: 'query', regex: /^[A-z0-9\s\-\'\.]+$/i, description: 'crafting material to search for, use quotes for multiple words' }
+        ]
+      ]
+    args = parse_args(arg_definitions)
+    container = get_settings.crafting_container
+    
+    stow_hands
+    if get_item('deed register', container)
+	  bput("turn my register to contents", 'You flip your deed register to the contents page.', 'But a deed register is already at the table of contents!')
+      search(args.query)
+	else
+	  echo "Can\'t find your deed register."
+	  exit
+	end
+    stow_hands
+  end
+
+  def display_results(results)
+    results.each do |data|
+      respond(data)
+    end
+  end
+
+  def search(query)
+    contents = []
+    bput("read my register", 'Stored Deeds:')
+    pause
+    loop do
+	  line = get?
+	  if line.match("You haven't stored any deeds in this register.")
+	    contents << "   There are no deeds in your register."
+		break
+	  end
+	  if line.match("You shouldn't read somebody else's deed book.")
+	    contents << "   This is not your deed register!"
+		break
+	  end
+	  if line.match(/\W#{query}\W/i)
```

Comment:
A regex of `\Wlumber\W` _requires_ there to be a non-word character after the word `lumber`. You can use `\W*` if you want _zero or more_ non-word characters.

Reply:
Thanks, it works properly now.